File: Grind75/twoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int>uMap;
        for (int i = 0; i < nums.size(); i++) {
            int element = target - nums[i];
            if (uMap.find(element) != uMap.end()) {
                return {uMap[element], i};
            }
            uMap[nums[i]] = i;
        }
        return {-1,-1};
    }
};

File: twoSum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        uMap = {}
        for i in range(len(nums)):
            value = target - nums[i]
            if value in uMap:
                return [i, uMap[value]]
            uMap[nums[i]] = i
        
        return [-1,-1]
    
solution = Solution()
target = int(input())
nums = list(map(int,input().split()))
res = solution.twoSum(nums, target)
print(res)
